template-vue-ts/src/components/HelloWorld.vue

```diff
@@ -0,0 +1,20 @@
+<template>
+  <h1>{{ msg }}</h1>
+  <button @click="count++">count is: {{ count }}</button>
+  <p>
+    Edit <code>components/HelloWorld.vue</code> to test hot module replacement.
+  </p>
+</template>
+
+<script lang="ts">
+import { defineComponent, ref } from 'vue'
+
+export default defineComponent({
+  name: 'HelloWorld',
+  setup: (props: { readonly msg: string }) => {
+    const { msg } = prop
```

Comment:
1. `msg` must be declared via `props` option
2. With declared props you don't need the type annotation for `props` in `setup`
3. The destructuring doesn't do anything - in fact, because props are auto exposed on render context when declared, `setup` doesn't even need to do anything about it. Destructuring here also breaks props reactivity.

Reply:
The declaration through props option looks redundant, because this is not js code or component library, but ts code. But I will change it according to this.